sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -146,6 +151,15 @@ case class CatalogTable(
   requireSubsetOfSchema(sortColumnNames, "sort")
   requireSubsetOfSchema(bucketColumnNames, "bucket")
 
+  lazy val userSpecifiedSchema: Option[StructType] = if (schema.nonEmpty) {
```

Comment:
oh, having this is because `CatalogColumn` is using string as the type? I think we should just use `StructType` as the schema and remove `CatalogColumn`?